src/Illuminate/Session/Middleware/StartSession.php

```diff
@@ -243,4 +252,16 @@ protected function usingCookieSessions()
 
         return $this->manager->driver()->getHandler() instanceof CookieSessionHandler;
     }
+
+    /**
+     * Determines if the specified session needs garbage collection for expired
+     * sessions.
+     *
+     * @param  \Illuminate\Session\SessionInterface  $session
+     * @return bool
+     */
+    protected function gcNeeded(SessionInterface $session)
+    {
+        return ! ($session->getHandler() instanceof CacheBasedSessionHandler);
```

Comment:
> We should have something more explicit than this.

@JosephSilber You mean sth. like a class property on the handler indicating if the handler class needs gc or not? If you mean that then yes, I would agree but I guess anything beyond the current changes would be breaking changes imo.

> don't need the bracks here either

@GrahamCampbell Imo the length of the expression followed by the `!` negation justifies the brackets because it makes the conditional statement more readable and expressive.
I understand that everyone has their own programming style and that there might be a very stringent code style guideline but it is kind of time consuming changing all those little things in PRs. Don't you have a style guide listing all those little things beyond PSR-2 ?